storages/python/python-tests/postgresql_test.py

```diff
@@ -0,0 +1,275 @@
+
+
+if __name__ in ('__main__', '<run_path>'):
+    print('Running Python test <postgresql_test>...')
+    thisdir = Path(__file__).absolute().parent
+    input_path = thisdir / 'input'
+    dlite_path = thisdir.parent.parent.parent
+    plugin = thisdir.parent / 'python-storage-plugins/postgresql.py'
+    
+    try:
+        with open(plugin, 'r') as orig:
+            lines = orig.read().splitlines(keepends=True)
+            # Alter the plugin methods open(), load(), save() and
+            # close(), to use files for comparison, instead
+            # of databases contained in a PostgreSQL server.
+            # This removes the need for a server when running the test.
+            #
+            # The input comparisons are between JSON files and
+            # PostgreSQL database dumps:
+            # 1: 'test-entity.json' vs. '/input/test_meta.pgsql'
+            # 2: 'test-data.json' vs. '/input/test_data.pgsql'
+            #
+            # The output comparisons are between SQL commands from the
+            # save() method (that would have been executed on a server)
+            # and text files containing the desired commands:
+            # 1: '/input/postgresql_test_meta_save.txt'
+            # 2: '/input/postgresql_test_data_save.txt'
+            
+            df = '    def '
+            ind8 = '        ' # Indent of 8 spaces
+            open_start = lines.index(df + 'open(self, uri, options=None):\n')
+            close_start = lines.index(df + 'close(self):\n')
+            load_start = lines.index(df + 'load(self, uuid):\n')
+            save_start = lines.index(df + 'save(self, inst):\n')
+            
+            # open(): Don't connect to server - read 'uri' instead
+            lines[open_start + 1] = ind8 + 'self.data = open_pgsql(uri)\n'
+            lines[open_start + 2] = ind8 + 'self.d = {}\n'
+            
+            # close(): Don't disconnect from server - just pass
+            lines[close_start + 1] = ind8 + 'pass\n'
+            
+            # load(): Don't access server - read from self.data
+            lines[load_start + 3] = ind8 + 'self.d[uuid] = ' \
+                + 'load_pgsql(self.data, uuid, ["L", "M", "N"])\n'
+            lines[load_start + 4] = ind8 \
+                + 'return instance_from_dict(self.d[uuid])\n'
+            
+            # save(): Don't write to database, but compare the writing
+            # commands to the commands in the database dump file
+            n = save_start + 2
+            lines[n - 1] = ind8 + 'ret = {"uuid": inst.uuid}\n'
+            while not lines[n].startswith(df + 'table_exists'):
+                lines[n] = lines[n].replace('self.conn', '#')
+                lines[n] = lines[n].replace('self.cur.execute(', \
+                    'ret = extract_exec_args(ret, ')
+                if lines[n].startswith(ind8 + 'if not self.table'):
+                    lines[n] = '\n'
+                    lines[n + 1] = '\n'
+                n = n + 1
```

Comment:
```suggestion
                n += 1
```
Generally preferable, since it avoids generating a new object